### PR TITLE
Patch `GLGraphicsItem` and `GLViewMixin` to support nested `GLLayer`s

### PR DESCRIPTION
**Pull request by williamckha**
### Description

- Patch `GLGraphicsItem.setParentItem`, `GLViewMixin.addItem`, and `GLViewMixin.removeItem` to ensure that `GLGraphicsItem`s are always added to the view, regardless of how deep they are in the scenegraph hierarchy (currently only top-level `GLGraphicsItem`s are added to the view)

- Support nesting of layers (i.e. `GLLayer`s can be added as children of another `GLLayer`)

- Switch to using `pyqtdarktheme` (I will try to implement a settings menu so that we can switch between different themes). Using `pyqtdarktheme-fork` since it supports python 3.12

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

**Review by itsarune — APPROVED** [review]
Awesome! Is this a known pyqtgraph issue?

`pyqtdarktheme` works on my computer. It would be nice if someone with an ARM machine can also cross-check.

**Comment by williamckha** [comment]
> Awesome! Is this a known pyqtgraph issue?

I dunno, but they've mostly stopped development on the 3D library in pyqtgraph. This also might not be the way they intended us to use the 3D library API

**Review by Muxite — COMMENTED** [review]
very cool

File: environment_setup/ubuntu24_requirements.txt [hunk 1]
```diff
@@ -4,5 +4,4 @@ iterfzf==0.5.0.20.0
 python-Levenshtein==0.25.1
 psutil==5.9.0
 PyOpenGL==3.1.6
-qt-material==2.12
 ruff==0.5.5
```
Comment (Mr-Anyone) [inline environment_setup/ubuntu24_requirements.txt:7]: do we need to put this here for `setup_software.sh` to work? 
```pyqtdarktheme-fork```

File: src/software/thunderscope/gl/helpers/gl_patches.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+        for child in item_to_remove.childItems():
+            items_to_remove.append(child)
+
+    self.update()
+
+
+GLGraphicsItem.setParentItem = GLGraphicsItem_setParentItem_patched
+GLViewMixin.addItem = GLViewMixin_addItem_patched
+GLViewMixin.removeItem = GLViewMixin_removeItem_patched
```
Comment (Mr-Anyone) [inline src/software/thunderscope/gl/helpers/gl_patches.py:71]: Did not know you could do something like that in python. 

Also, does the order of importing now matter? 

So will 

```
import <long_list_of_stuff>/gl_patches 
import pyqtgraph
```

verus
```
import pyqtgraph
import <long_list_of_stuff>/gl_patches 
```
matter?

**Review by Mr-Anyone — APPROVED** [review]
overall look good.

File: environment_setup/ubuntu24_requirements.txt (hunk 1 above)
Comment (williamckha) [reply environment_setup/ubuntu24_requirements.txt:7]: No

File: src/software/thunderscope/gl/helpers/gl_patches.py (hunk 2 above)
Comment (williamckha) [reply src/software/thunderscope/gl/helpers/gl_patches.py:71]: Problem not something you need to worry about. As long as the `import ... gl_patches` happens before the `GLViewWidget` or any `GLGraphicsItem`s are created, the patch should work

**Pull request merged**